core/src/utils.rs

```diff
@@ -0,0 +1,33 @@
+use std::time::Duration;
+
+use log::debug;
+use solana_sdk::commitment_config::CommitmentConfig;
+use tokio::time::{timeout, Instant};
+
+use crate::{structures::block_info::BlockInfo, types::BlockInfoStream};
+
+pub async fn get_latest_block_info(
+    mut blockinfo_stream: BlockInfoStream,
+    commitment_config: CommitmentConfig,
+) -> BlockInfo {
+    let started = Instant::now();
+    loop {
+        match timeout(Duration::from_millis(500), blockinfo_stream.recv()).await {
+            Ok(Ok(block_info)) => {
+                if block_info.commitment_config == commitment_config {
+                    return block_info;
+                }
+            }
+            Err(_elapsed) => {
+                debug!(
+                    "waiting for latest block info ({}) ... {:.02}ms",
+                    commitment_config.commitment,
+                    started.elapsed().as_secs_f32() * 1000.0
+                );
+            }
+            Ok(Err(_error)) => {
```

Comment:
**!!** you should handle Lagged and Closed  are differently:
* lagged: continue
* Closed: sleep some 10 second, then panic (create_grpc_multiplex_blocks_subscription _reconnect_loop_ never terminates unless cluster_endpoint_tasks get aborted which happens on shutdown; this method could gracefully wait if it detects shutdown)

Reply:
The aim of this method is to wait till we get block of required commitment, this channel should never lag. If it lags then the logic is wrong.